packages/engine-server/src/workspace/service.ts

```diff
@@ -527,10 +519,63 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
       });
     }
 
+    const config = this.config;
+    ConfigUtils.updateVault(config, targetVault, (vault) => {
+      vault.remote = {
+        type: "git",
+        url: remoteUrl,
+      };
+      return vault;
+    });
+
+    let ignorePath: string = targetVault.fsPath;
+    if (config.dev?.enableSelfContainedVaults) {
+      // Move vault folder to the correct location
+      const vaultName =
+        targetVault.name ??
+        // if the vault has no name, compute one based on the path
+        _.findLast(
```

Comment:
do we need to remove trailing slash in case the path ends with one? `eg. foo/ or foo\`

Reply:
If you're asking about 'findLast`, it's filtering for segments with non-0 length so it can handle a trailing slash (or even multiple trailing slashes like `foo////`

Reply:
got it, disregard then